src/timestamp.ts

```diff
@@ -1,15 +1,24 @@
 import { Long } from './long';
 
-/**
- * @class
- * @param {number} low  the low (signed) 32 bits of the Timestamp.
- * @param {number} high the high (signed) 32 bits of the Timestamp.
- * @return {Timestamp}
- */
-export class Timestamp extends Long {
-  _bsontype: string;
+type TimestampOverrides = '_bsontype' | 'toExtendedJSON' | 'fromExtendedJSON';
+type LongWithoutOverrides = new (low: number | Long, high?: number, unsigned?: boolean) => {
+  [P in Exclude<keyof Long, TimestampOverrides>]: Long[P];
+};
+const LongWithoutOverridesClass: LongWithoutOverrides = (Long as unknown) as LongWithoutOverrides;
 
+export class Timestamp extends LongWithoutOverridesClass {
+  _bsontype!: 'Timestamp';
```

Comment:
I know what you're thinking.. this looks gross! 😅 
This is the only case where we extend from another BSON type but this means we can't just have different types for `_bsontype` and from/toExtendedJSON so this does some trickery to create a Long type that doesn't have those properties. The generated code is not effected it still extends Long as normal.

Reply:
Another way to accomplish this would be to make something like:
```
class LongBase { ... the vendored Long implementation ... }

export class Long extends LongBase {
  _bsontype!: 'Long';
  fromExtendedJSON(..) {}
  toExtendedJSON(..) {}
}

export class Timestamp extends LongBase {
  _bsontype!: 'Timestamp';
  fromExtendedJSON(..) {}
  toExtendedJSON(..) {}
}
```

This approach might make it easier to use `BigInt` in the future as well?

Reply:
This is a good solution!
But there's a type issues that arise in a few places, namely Long.isLong asserts that something is `LongBase` and then we would need a further check to see if something is Long with _bsontype
My solution lives only in TS land and doesn't effect code output. I know its a hack kinda but it feels like the right way to go to keep types consistent.

Reply:
looks like the [definitely typed typings](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/bson/index.d.ts#L210) use a similar approach to what I suggested?

Reply:
Oh yea I could paramertize the Long class, I'll look into that.